appium/webdriver/extensions/flutter_integration/flutter_commands.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import os
+from typing import Any, Dict, Optional, Tuple, Union
+
+from appium.webdriver.extensions.flutter_integration.scroll_directions import ScrollDirection
+from appium.webdriver.flutter_finder import FlutterFinder
+from appium.webdriver.webdriver import WebDriver
+from appium.webdriver.webelement import WebElement
+
+
+class FlutterCommand:
+
+    def __init__(self, driver: WebDriver) -> None:
+        self.driver = driver
+
+    # wait commands
+
+    def wait_for_visible(self, locator: Union[WebElement, FlutterFinder], time_out: Optional[float] = None) -> None:
+        """
+        Waits for a element to become visible.
+
+        Args:
+            locator (Union[WebElement, FlutterFinder]): The element to wait for; can be a WebElement or a FlutterFinder.
+            time_out (Optional[float]): Maximum wait time in seconds. Defaults to a predefined timeout if not specified.
+
+        Returns:
+            None:
+        """
+        opts: Dict[str, Union[WebElement, Dict[str, str], float]] = {}
+        if isinstance(locator, WebElement):
```

Comment:
this condition is duplicated multiple times. Please extract it to a function

Reply:
Thanks updated